ql/math/array.hpp

```diff
@@ -463,6 +464,14 @@ namespace QuantLib {
         return reverse_iterator(begin());
     }
 
+    inline void Array::resize(Size n) {
+        if (n != n_) {
```

Comment:
If `n < n_`, I would avoid the allocation and simply change `n_`.  We'll have some unused storage, but that's ok, I guess.